api/v1alpha3/conversion.go

```diff
@@ -118,3 +117,15 @@ func Convert_v1alpha3_OpenStackClusterSpec_To_v1alpha4_OpenStackClusterSpec(in *
 func Convert_v1alpha3_OpenStackMachineSpec_To_v1alpha4_OpenStackMachineSpec(in *OpenStackMachineSpec, out *v1alpha4.OpenStackMachineSpec, s conversion.Scope) error {
 	return autoConvert_v1alpha3_OpenStackMachineSpec_To_v1alpha4_OpenStackMachineSpec(in, out, s)
 }
+
+// Convert_v1alpha3_OpenStackClusterSpec_To_v1alpha4_OpenStackClusterSpec has to be added by us for the new portOpts
+// parameter in v1alpha4. There is no intention to support this parameter in v1alpha3, so the field is just dropped.
+func Convert_v1alpha4_Network_To_v1alpha3_Network(in *v1alpha4.Network, out *Network, s conversion.Scope) error {
```

Comment:
I think this should be like it says in the comment `Convert_v1alpha3_OpenStackClusterSpec_To_v1alpha4_OpenStackClusterSpec`, I think we don't need specific conversion function for Network since it is part of the `OpenStackMachineSpec`.

I am pretty sure when changing between api versions you just can't drop any fields that are not supported in another. You need to store and restore these field during conversion.

Reply:
I see that we'll drop fields if we go v1alpha4 -> v1alpha3 -> v1alpha4. This will also apply to other fields added/removed in v1alpha4 that aren't part of this change (e.g. .UserDataSecret, useOctavia). Can we raise and fix robust conversion between versions as a separate issue outside the scope of this change?

Happy to accept any suggestions on improving or fixing these conversion functions that I have written, because I have largely guessed at what they are supposed to be in order to get the code generation passing without warnings or errors.